### ISSUE TITLE
Multithreading is plattform dependent

### ISSUE THREAD

**Issue by chraibi** [JuPedSim/jpsvis#66]
In Gitlab by @chraibi on Mar 28, 2019, 18:11 [[origin](https://gitlab.version.fz-juelich.de/jupedsim/jpsvis/issues/67)]

In MainWindow.cpp we handle multithreading differently, according to the used platform.  For example:

```
#ifdef __APPLE__
            //std::thread::id main_thread_id = std::this_thread::get_id();
            dispatch_queue_t main_q = dispatch_get_main_queue();
            dispatch_async(main_q, ^(void){
                               //visualisationThread->moveToThread(QApplication::instance()->thread());
                               _visualisationThread->run();
                           });
#else
            _visualisationThread->start();

#endif
```

here for Macosx, we use  Apple's threading system which is built right into the operating system.

I'm not sure if using POSIX threads or even Boost:Threads is a better option in terms of performance.

**Comment by chraibi** [comment]
In Gitlab by @chraibi on Mar 28, 2019, 18:58

created branch [`67-multithreading-is-plattform-dependent`](https://gitlab.version.fz-juelich.de/jupedsim/jpsvis/compare/master...67-multithreading-is-plattform-dependent) to address this issue

**Comment by Ozaq** [comment]
This is fixed with #96
